src/workspaces/useWorkspaces.ts

```diff
@@ -0,0 +1,15 @@
+import { workspaceProvider } from 'src/libs/ajax/workspaces/providers/WorkspaceProvider';
+import { useSettableStore } from 'src/libs/react-utils';
+import { workspacesStore } from 'src/libs/state';
+import { makeUseWorkspaces } from 'src/workspaces/useWorkspaces.composable';
+
+export const useWorkspacesState = () => useSettableStore(workspacesStore);
+
+export const useWorkspacesComposer = {
+  make: () =>
+    makeUseWorkspaces({
+      workspaceProvider,
+      useWorkspacesState,
+    }),
+};
+export const useWorkspaces = useWorkspacesComposer.make();
```

Comment:
What's the purpose of `useWorkspacesComposer`? As opposed to:

```ts
export const useWorkspaces = makeUseWorkspaces({
  workspaceProvider,
  useWorkspacesState,
});
```

Reply:
I don't see the value in the "is composed for Terra UI app" test, so I'd still vote for the simpler call here.

Reply:
there are cases where "would be really bad if this broke" is the top reason for a test... but this will be reworked based on other feedback.

Reply:
Sure. But if I look at a failed test and see that it's testing the exact implementation of something, I'm probably going to update the test instead of thinking it's actually identifying an issue. If changing _anything_ in a function requires a change to a test, I'm skeptical of the usefulness of that test.

Reply:
test reworked